### PR TITLE
server: add region leader for region api.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: server/cache.go [hunk 1]
```diff
@@ -188,17 +188,23 @@ func (r *regionsInfo) getRegion(regionKey []byte) *metapb.Region {
 	return nil
 }
 
-// getRegionByID gets the region by regionID, returns nil if not found.
-func (r *regionsInfo) getRegionByID(regionID uint64) *metapb.Region {
+// getLeaderRegion gets the region and leader peer by regionID.
+func (r *regionsInfo) getLeaderRegion(regionID uint64) (*metapb.Region, *metapb.Peer) {
```
Comment (shenli) [inline server/cache.go]: getRegionAndLeader is better

Comment (qiuyesuifeng) [inline server/cache.go]: Yes, that is better.

**Comment by qiuyesuifeng** [comment]
@shenli  PTAL

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
@siddontang PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**